### PR TITLE
Remove foreign fetch

### PR DESCRIPTION
**Pull request by jakearchibald**
Removing the foreign fetch parts. I've left the <link rel=serviceworker> stuff in there for now, we may discuss that at TPAC.

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/ServiceWorker/remove-foreign-fetch.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/ServiceWorker/0bbf95c...781bef8.html)

### CODE REVIEW

**Comment by jakearchibald** [comment]
Creating a PR to remove the tests…

**Comment by jakearchibald** [comment]
https://github.com/w3c/web-platform-tests/pull/7762

File: docs/index.bs [hunk 1]
```diff
@@ -1725,11 +1557,6 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
       </thead>
       <tbody>
         <tr>
-          <td><dfn event id="service-worker-global-scope-install-event"><code>install</code></dfn></td>
```
Comment (mkruisselbrink) [inline docs/index.bs]: This block should still be there, right? Just with ExtendableEvent rather than InstallEvent

**Comment by jakearchibald** [comment]
Hah, I forgot to escape the html, I meant the `<link rel=serviceworker>` stuff

File: docs/index.bs (hunk 1 above)
Comment (jakearchibald) [reply docs/index.bs]: Ah yes, correct

**Comment by jakearchibald** [comment]
Before I land this… should we keep `InstallEvent` now we have it? Or just reintroduce it later when we need it?

**Comment by wanderview** [comment]
> Before I land this… should we keep InstallEvent now we have it? Or just reintroduce it later when we need it?

Currently we have the weird thing in `waitUntil()` steps that does special logic for install:

https://w3c.github.io/ServiceWorker/#wait-until-method

If we kept `InstallEvent` we could move that to an `InstallEvent.waitUntil()` that implements that behavior instead of special casing it ExtendableEvent.  Not sure if thats worth it or not.

**Comment by jakearchibald** [comment]
F2F: Keeping the header (for navigation requests) is interesting to CDNs. We'll talk to them this week and find out how important it is.

Going back to `ExtendableEvent` for install seems fine. Mozilla still uses `ExtendableEvent`.

**Comment by jungkees** [comment]
> If we kept InstallEvent we could move that to an InstallEvent.waitUntil() that implements that behavior instead of special casing it ExtendableEvent. Not sure if thats worth it or not.

That behavior's already spec'ed in Install algorithm and Activate algorithm. I think we can safely change the prose in the waitUntil section to notes. Filed https://github.com/w3c/ServiceWorker/issues/1227.

So, we can just use `ExtendableEvent` as discussed in the F2F.

**Comment by jakearchibald** [comment]
I've removed the link header & tag, but we should talk more to CDNs and add it again if needed.

Still TODO: remove from tests and HTML spec.

**Comment by jakearchibald** [comment]
HTML PR: https://github.com/whatwg/html/pull/3233
Tests PR: https://github.com/w3c/web-platform-tests/pull/7762

**Pull request merged**